rhoas/kafkas/resource_kafka.go

```diff
@@ -122,6 +123,9 @@ func kafkaDelete(ctx context.Context, d *schema.ResourceData, m interface{}) dia
 		Refresh: func() (interface{}, string, error) {
 			data, resp, err1 := api.KafkaMgmt().GetKafkaById(ctx, d.Id()).Execute()
 			if err1 != nil {
+				if err1.Error() == "404 Not Found" {
+					return data, "404", nil
+				}
```

Comment:
Why are we checking this just not sure? Also we will do more error handling in the future anyway so nothing needs to be perfect right now.

Reply:
Similarly to [this comment|https://github.com/redhat-developer/terraform-provider-rhoas/pull/24#discussion_r982093684], I discovered that sometimes the API returns a "404 Not Found" and the delete function errors. I think that if we get a 404, there shouldn't be any error, as not finding the resource means it has been deleted.